### ISSUE TITLE
Lambda invoke function is not mocked

### ISSUE THREAD

**Issue by mikegrima** [spulec/moto#629]
Hello:

I am attempting to mock out a call to the AWS Lambda `invoke()` function.  Instead of hitting Moto, it appears to be reaching out to AWS directly and failing as a result:

```
self = <botocore.retryhandler.ExceptionRaiser object at 0x10349b090>
attempt_number = 5
caught_exception = ConnectionClosedError(u'Connection was closed before we received a valid respo...lambda.us-west-2.amazonaws.com/2015-03-31/functions/MYFUNCTION/invocations".',)

    def _check_caught_exception(self, attempt_number, caught_exception):
        # This is implementation specific, but this class is useful by
        # coordinating with the MaxAttemptsDecorator.
        # The MaxAttemptsDecorator has a list of exceptions it should catch
        # and retry, but something needs to come along and actually raise the
        # caught_exception.  That's what this class is being used for.  If
        # the MaxAttemptsDecorator is not interested in retrying the exception
        # then this exception just propogates out past the retry code.
>       raise caught_exception
E       ConnectionClosedError: Connection was closed before we received a valid response from endpoint URL: "https://lambda.us-west-2.amazonaws.com/2015-03-31/functions/MYFUNCTION/invocations".
```

I was looking at the moto code for Lambda, and wasn't quite sure where the issue lies or how to fix it.  I recall making a PR in the past, and for that PR, I recall adding in the XML response returned from AWS.  Things appear to be very different with the newer AWS APIs, so I'm not entirely sure how to implement moto fixes now...

I was looking to see if adding an `invoke` method to https://github.com/spulec/moto/blob/master/moto/awslambda/models.py would work, but it doesn't seem to get executed.

Do you know what may be causing the `invoke` method to get skipped over, and where to place the code to fix it?

Thank You

**Comment by spulec** [comment]
It looks like invocation hasn't been added yet.

We would need to add the [url endpoint](https://github.com/spulec/moto/blob/master/moto/awslambda/urls.py), map it to a new [response]( https://github.com/spulec/moto/blob/master/moto/awslambda/responses.py), and add a method on the [lambda function model](https://github.com/spulec/moto/blob/master/moto/awslambda/models.py) to execute.

I probably won't have time for this in the near future, but would be happy to accept a PR if anyone has time.

**Comment by mikegrima** [comment]
Hi @spulec I wouldn't mind submitting a PR for now.  Thank you for providing some insight on how it works.

I'll let you know if I hit any stumbling blocks.

**Comment by mikegrima** [comment]
Making good progress... Almost done.

**Comment by mikegrima** [comment]
PR submitted.  Not sure why a bunch of other tests fail.

**Comment by spulec** [comment]
Closed with #644
